src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemorySegment.java

```diff
@@ -1335,7 +1335,8 @@ default void set(ValueLayout.OfAddress layout, long offset, Addressable value) {
      */
     @ForceInline
     default char getAtIndex(ValueLayout.OfChar layout, long index) {
```

Comment:
We need to thread lightly here. The alignment check is not free, performance-wise, because of [1]. So, if we can detect that the var handle will always be accessed at offsets that are multiple of the alignment, we can use the more optimized "aligned" var handle, which will skip the alignment check. Eventually this special casing will go away when [1] is fixed.